core/shared/src/main/scala/zio/prelude/NonEmptyList.scala

```diff
@@ -78,6 +78,21 @@ sealed trait NonEmptyList[+A] { self =>
       if (seen(a)) (as, seen) else (cons(a, as), seen + a)
     }._1.reverse
 
+  /** Decomposes the `NonEmptyList` into an element and a (possibly empty) `List` */
+  final def destruct: (A, List[A]) = self match {
```

Comment:
Isn't this just pattern matching on the list?

Reply:
Yep, but it is sometimes handy to have it like a method.